### DISCUSSION TITLE
Create subplots over groupby rather then by

### DISCUSSION THREAD

**Discussion by jsignell** [holoviz/hvplot#1572] [General]
I am not sure how the ambiguity of this should be handled, but I have a plot with both a `groupby` and a `by`. I don't want to have a selector widget so I'd like to be able to get subplots using `subplots=True`. But this splits into subplots over the `by` field rather than the `groupby`

<img width="1230" alt="screen shot 2019-01-31 at 1 13 45 pm" src="https://user-images.githubusercontent.com/4806877/52075367-15878c80-255a-11e9-9374-5991950e9b21.png">

### To Reproduce:
```python
import pandas as pd
import hvplot.pandas

df = pd.read_csv('benchmarking.txt')
df['total_cores'] = df['njobs'] * df['ncores']

df.hvplot('total_cores', 'seconds', by='nworkers', groupby='operation', subplots=True,
                width=400, height=400, ylim=(0,1))
```
[benchmarking.txt](https://github.com/pyviz/hvplot/files/2818504/benchmarking.txt)

**Comment by jbednar** [comment]
I admit I've been confused between `by` and `groupby`...

**Comment by jsignell** [comment]
In general `by` creates overlays and `groupby` creates selectors. The work-around that Philipp suggests is to use call `.layout()` on the output:

<img width="1177" alt="screen shot 2019-01-31 at 1 19 13 pm" src="https://user-images.githubusercontent.com/4806877/52075675-defe4180-255a-11e9-9d98-6f9c96a52cab.png">

**Comment by philippjfr** [comment]
@tonyfast suggests that we could alias `by` with `hue` which is used by seaborn and is a lot clearer.

**Comment by philippjfr** [comment]
@holoviz/triaging-team any thoughts?

**Comment by jbednar** [comment]
No objection; just need to document that `hue` is the same as `by`...

**Comment by maximlt** [comment]
I've also been confused by `by` and `groupby`. @philippjfr @jbednar still in favor of aliasing `by` with `hue`? It sounds like a good idea.

**Comment by maximlt** [comment]
xarray plotting also uses `hue`.

**Comment by droumis** [comment]
Our discussion on discord is leaning toward the use of `colorby` instead of hue

**Comment by philippjfr** [comment]
Tbh, I have some issues with both `colorby` and `hue` because we already support `color=<dimension≥`. The current `by` operation effectively maps onto an NdOverlay in HoloViews. Most users won't care about that but it's still conceptually different from simply coloring by a dimensions, it just so happens that we define a color cycle for layers in an overlay. Sadly I don't have any good suggestions to communicate the subtleties.
